packages/jest-schemas/src/raw-types.ts

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* eslint-disable sort-keys */
+
+import {Type} from '@sinclair/typebox';
+
+export const RawSnapshotFormat = Type.Partial(
+  Type.Object({
+    callToJSON: Type.Boolean(),
+    compareKeys: Type.Null(),
+    escapeRegex: Type.Boolean(),
+    escapeString: Type.Boolean(),
+    highlight: Type.Boolean(),
+    indent: Type.Integer({minimum: 0}),
+    maxDepth: Type.Integer({minimum: 0}),
+    maxWidth: Type.Integer({minimum: 0}),
+    min: Type.Boolean(),
+    printBasicPrototype: Type.Boolean(),
+    printFunctionName: Type.Boolean(),
+    theme: Type.Partial(
+      Type.Object({
+        comment: Type.String(),
+        content: Type.String(),
+        prop: Type.String(),
+        tag: Type.String(),
+        value: Type.String(),
+      }),
+    ),
+  }),
+);
+
+const RawCoverageProvider = Type.Union([
+  Type.Literal('babel'),
+  Type.Literal('v8'),
+]);
+
+const RawCoverageThresholdValue = Type.Partial(
+  Type.Object({
+    branches: Type.Number({minimum: 0, maximum: 100}),
+    functions: Type.Number({minimum: 0, maximum: 100}),
+    lines: Type.Number({minimum: 0, maximum: 100}),
+    statements: Type.Number({minimum: 0, maximum: 100}),
+  }),
+);
+
+const RawCoverageThreshold = Type.Intersect([
+  Type.Object({
+    global: RawCoverageThresholdValue,
+  }),
+  // TODO: is there a better way of doing index type?
```

Comment:
this is currently

```ts
type CoverageThreshold = {
  [path: string]: CoverageThresholdValue;
  global: CoverageThresholdValue;
};
```

I don't think my current approach is the correct "translation" of that, but it seems to work in practice

Reply:
Hi @SimenB!

The above type would represent the TS data structure (both validation and inference). However the allOf representation could be a consideration if publishing (as it would require downstream tooling to be able to enumerate properties of the intersection)

There is another way to represent this type, which would ensure it remains an "object" type.

```typescript
const RawCoverageThreshold = Type.Object({
  global: RawCoverageThresholdValue,              // global: RawCoverageThresholdValue
}, {
  additionalProperties: RawCoverageThresholdValue // [path: string]: RawCoverageThresholdValue 
})
```
TypeBox does not support auto inference for `additionalProperties` of `TSchema`, But it is possible to wrap in Unsafe which allows you to specify the correct inference type.

```typescript
const RawCoverageThresholdBase = Type.Object({
  global: RawCoverageThresholdValue,
}, {
  additionalProperties: RawCoverageThresholdValue
})

const RawCoverageThreshold = Type.Unsafe<{
  global: Static<typeof RawCoverageThresholdValue>,
  [path: string]: Static<typeof RawCoverageThresholdValue>
}>(RawCoverageThresholdBase)
```
It's also possible to pass Unsafe arbitrary JSON Schema + Type if you need specific representations outside the ones provided by TypeBox.

Hope this helps! :)

Reply:
it does, thank you!